### ISSUE TITLE
MD076: Allow option for list continuation items to be loose in an otherwise tight list

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#524]
### Problem or motivation

MD076 (https://github.com/rvben/rumdl/blob/a00435eea5c100673c59a99e3e434825992a90c1/docs/md076.md) is a useful rule for keeping list items together when used in "tight mode". List items that belong together in the same list can't have spaces between them, which can look at first glance like the start of a new list.

However, list continuation text counts as a list item, in some cases, on whether a list is properly "tight" or "loose". For example:

This list is completely tight:

```markdown
- List item.
  List continuation item.
- List item.
  List continuation item.
- List item.
- List item.
- List item.
  List continuation item.
- List item.
- List item.
```

This is combines tight and loose elements, but is still considered tight by `rumdl` (no violations of MD076 tight style is noted):

```markdown
- List item.

  List continuation item.
- List item.

  List continuation item.
- List item.
- List item.
- List item.

  List continuation item.
- List item.
- List item.
```

The reason, I believe, why this second list is still considered tight by `rumdl` is because the list continuation items still unambiguously belong to the list items that precedes them.

However, folks may want the option to have this style so that list continuation content stands out more:

```markdown
---
title: Heading
---

- List item.

  List continuation item.

- List item.

  List continuation item.

- List item.
- List item.
- List item.

  List continuation item.

- List item.
- List item.
```

I think the reason why this fails MD076 right now is because it's hard to tell if a new list is intended anywhere here, or if this should all be one list.

I wonder, though, if folks should be able to opt-out of "strict mode", where the behaviour for list continuation text is more relaxed than for list item text. Loose mode is permitted now for folks that want it, but some folks might want tightness for main list items, and looseness for list continuation text.

### Proposed solution

Provide an additional setting for list continuation text for MD076: https://github.com/rvben/rumdl/blob/a00435eea5c100673c59a99e3e434825992a90c1/docs/md076.md#configuration.

Something like:

```toml
[MD076]
style = ["consistent", "tight, "loose"]
strict = boolean
```

To give an example:

- This configuration:
  ```toml
  [MD076]
  style = "tight"
  strict = true
  ```
- Would require this Markdown:
  ```markdown
  - List item.
    List continuation item.
  - List item.
    List continuation item.
  - List item.
  - List item.
  - List item.
    List continuation item.
  - List item.
  - List item.
  ```

Another example:

- This configuration:
  ```toml
  [MD076]
  style = "loose"
  strict = true
  ```
- Would require this Markdown:
  ```markdown
  - List item.

    List continuation item.

  - List item.

    List continuation item.

  - List item.

  - List item.

  - List item.

    List continuation item.

  - List item.

  - List item.
  ```

And finally:

- This configuration:
  ```toml
  [MD076]
  style = "tight"
  strict = false
  ```
- Would permit this style:
  ```markdown
  - List item.

    List continuation item.

  - List item.

    List continuation item.

  - List item.
  - List item.
  - List item.

    List continuation item.

  - List item.
  - List item.
  ```

### Alternatives considered

An alternative would be to change `strict` into `list-continuation`, and rather than have it be boolean, it could be `tight` or `loose`. For example, to set very tight lists:

```toml
[MD076]
style = "tight"
list-continuation = "tight"
```

To set very loose lists:

```toml
[MD076]
style = "loose"
list-continuation = "loose"
```

To set lists that are tight for main items, and loose for list continuation items:

```toml
[MD076]
style = "tight"
list-continuation = "loose"
```

### Feature category

Improve existing rule

### Example markdown

```markdown
---
title: Heading
---

- List item.

  List continuation item.

- List item.

  List continuation item.

- List item.
- List item.
- List item.

  List continuation item.

- List item.
- List item.
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

### Additional context

This feature request is similar to: https://github.com/rvben/rumdl/issues/523, but more generic.

**Comment by rvben** [comment]
Implemented in v0.1.50 with the `allow-loose-continuation` option:

```toml
[MD076]
style = "tight"
allow-loose-continuation = true
```

When enabled, blank lines around continuation paragraphs within list items are permitted even in tight mode. See [docs/md076.md](https://github.com/rvben/rumdl/blob/main/docs/md076.md#allow-loose-continuation) for details. Thanks for the suggestion, @eread!

**Comment by eread** [comment]
> Implemented in v0.1.50 with the `allow-loose-continuation` option:
> 
> [MD076]
> style = "tight"
> allow-loose-continuation = true
> When enabled, blank lines around continuation paragraphs within list items are permitted even in tight mode. See [docs/md076.md](https://github.com/rvben/rumdl/blob/main/docs/md076.md#allow-loose-continuation) for details. Thanks for the suggestion, [@eread](https://github.com/eread)!

Thanks @rvben. Works great!
